src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2720,6 +2796,7 @@ private BoundStatement BindReturn(ReturnStatementSyntax syntax, DiagnosticBag di
                     if (conversion.Kind != ConversionKind.Identity)
                     {
                         Error(diagnostics, ErrorCode.ERR_RefReturnMustHaveIdentityConversion, argument.Syntax, returnType);
+                        argument = argument.WithHasErrors();
```

Comment:
>argument = argument.WithHasErrors(); [](start = 24, length = 36)

why do we do this only in this case, but not other branches of this code that call `Error(...)`?

Reply:
To suppress the second diagnostic in `RefLocalsAndReturnsTests.BadRefReturnWrongType` which would otherwise report
``` none
                    // (6,20): error CS8151: The return expression must be of type 'int' because this method returns by reference
                    //         return ref i;
                    Diagnostic(ErrorCode.ERR_RefReturnMustHaveIdentityConversion, "i").WithArguments("int").WithLocation(6, 20),
                    // (6,20): error CS8156: An expression cannot be used in this context because it may not be passed or returned by reference
                    //         return ref i;
                    Diagnostic(ErrorCode.ERR_RefReturnLvalueExpected, "i").WithLocation(6, 20)
```
Ad for the other branches, we don't need a reason not to do something.  If cascaded diagnostic scenarios arise that prove that it useful, then we can add it.

---
In reply to: [301788345](https://github.com/dotnet/roslyn/pull/37052#discussion_r301788345) [](ancestors = 301788345)